src/registry.c

```diff
@@ -956,17 +971,14 @@ parse_layout(struct rxkb_context *ctx, xmlNode *layout,
         l = rxkb_layout_create(&ctx->base);
         list_init(&l->iso639s);
         list_init(&l->iso3166s);
-        l->name = name;
+        l->name = config.name;
         l->variant = NULL;
-        l->description = description;
-        l->brief = brief;
-        l->popularity = popularity;
+        l->description = config.description;
+        l->brief = config.brief;
+        l->popularity = config.popularity;
```

Comment:
we totally need a [g_steal_pointer](https://docs.gtk.org/glib/func.steal_pointer.html) or copy libei's [steal](https://gitlab.freedesktop.org/libinput/libei/-/blob/main/src/util-mem.h#L92) macro for the same effect. would allow us to just use an unconditional cleanup path.

Reply:
I will have a look, but let’s do this in another PR.

Reply:
I gave it a try in #407.